workspace/php2/src/XSolveSecurityBundle/Security/XsolveAuthenticationProviderManager.php

```diff
@@ -17,26 +18,33 @@ public function getUsernameForApiKey($apiKey) {
 
       $token = $this->entityManager->getRepository('XSolveSecurityBundle:Tokens')->findOneBy(['token' => $apiKey]);
       if (!$token) {
-	 throw new AccessDeniedException('unable to find token');
+
+	 throw new HttpException(401, "Unable to find token");
       }
+
       return $token->getUser()->getUsername();
    }
    public function loadUserByUsername($username) {
 
       $user = $this->entityManager->getRepository('XSolveSecurityBundle:User')->findOneBy(['username' => $username]);
       if (!$user) {
-	 throw new AccessDeniedException('unable to find user');
+
+	 throw new HttpException(401, "Unable to find user");
       }
+
       return $user;
    }
    public function refreshUser(UserInterface $user) {
      
       if (!$user instanceof WebserviceUser) {
```

Comment:
Już wiem! Powinno być       if (!$user instanceof User) {
Chodzi o to że powinno to sprawdzać czy user rzeczywiście należy do mojej stworzonej klasy użytkownika.